File: MoneyMoves/MoneyMoves/Data/Seed.swift
import Foundation

struct Quest: Identifiable, Hashable {
    let id: String
    let title: String
    let blurb: String
    let category: Category
    let xp: Int
    let coins: Int
    let lesson: [String]      // 3-5 short coach messages

    enum Category: String, Hashable {
        case paycheck, budgeting, investing, negotiation, credit, saving
        var emoji: String {
            switch self {
            case .paycheck:    return "💸"
            case .budgeting:   return "📊"
            case .investing:   return "📈"
            case .negotiation: return "🤝"
            case .credit:      return "💳"
            case .saving:      return "🐷"
            }
        }
    }
}

let QUESTS: [Quest] = [
    Quest(id: "paycheck", title: "Read your first paycheck",
          blurb: "Federal, state, FICA — where does it go?",
          category: .paycheck, xp: 60, coins: 30,
          lesson: [
            "Your gross is the sticker price — what you negotiated. Net is what hits your bank.",
            "FICA (7.65%) funds Social Security + Medicare. You can't avoid it but you'll see it forever.",
            "Federal + state income tax depend on your W-4. Too few withholdings = surprise bill in April.",
            "Pre-tax deductions (401k, HSA, health premiums) come OUT of gross BEFORE taxes — that's the magic of pre-tax saving.",
            "Action: open your last pay stub right now and find each line. You'll never look at a salary number the same way."
          ]),
    Quest(id: "budget-50-30-20", title: "The 50/30/20 starter budget",
          blurb: "Simple split: needs, wants, save.",
          category: .budgeting, xp: 80, coins: 40,
          lesson: [
            "50% needs (rent, food, transport, insurance), 30% wants, 20% save/invest. That's it.",
            "If 50% can't cover your needs — your fixed costs are too high. Roommate, cheaper area, transit.",
            "The 20% is sacred. It's the lever between you and your future self.",
            "Pro move: automate it. The day you get paid, 20% leaves before you can spend it.",
            "Action: pick a free tracker (Notion / Apple Numbers / your bank's tool) and log this month."
          ]),
    Quest(id: "first-roth", title: "Open a Roth IRA",
          blurb: "Tax-free growth. Best account in your 20s.",
          category: .investing, xp: 120, coins: 60,
          lesson: [
            "Roth IRA = pay tax NOW, grow tax-free, withdraw tax-free in retirement.",
            "Your tax rate at 22 (probably 12-22%) is likely lower than at 60. So Roth wins for most students.",
            "Limit is $7,000/year (2025). Even $50/month started at 22 = ~$200k at 65 at 7% returns.",
            "Open it at Fidelity / Schwab / Vanguard. Free, takes 10 min. Pick a target-date fund and forget.",
            "Action: bookmark the IRA application page. Even if you don't contribute today — you'll have the muscle memory."
          ]),
    Quest(id: "salary-anchor", title: "Anchor your salary ask",
          blurb: "The number you say first wins the room.",
          category: .negotiation, xp: 100, coins: 50,
          lesson: [
            "When recruiters ask 'what's your expectation' — they're betting you'll undersell.",
            "Counter: 'Based on the role and market data, I'd expect $X.' Pick X from levels.fyi for your level.",
            "Anchor high but defensible. Going 10-15% over their likely offer leaves room for both sides to feel good.",
            "If they offer below — silence. Then 'Can you walk me through how you got to that number?' Don't accept on the call.",
            "Action: practice your one-sentence anchor out loud. The words have to feel rehearsed."
          ]),
    Quest(id: "credit-myth", title: "The credit score myth",
          blurb: "What helps, what doesn't, what they don't tell you.",
          category: .credit, xp: 70, coins: 35,
          lesson: [
            "Credit score isn't about being responsible with money. It's about predictably paying lenders back.",
            "Biggest factor: payment history (35%). Never miss a payment. Even one 30-day late = -100 points.",
            "Second biggest: utilization (30%). Keep balances under 30% of your credit limit. 10% is even better.",
            "Closing old cards = bad (shortens average age). Asking for a higher limit = good (lowers utilization).",
            "Action: check your free FICO score (CreditKarma / Experian). Note your highest two factors and what's dragging you."
          ]),
    Quest(id: "emergency-fund", title: "Build a 3-month buffer",
          blurb: "The dumbest, most powerful financial move.",
          category: .saving, xp: 90, coins: 45,
          lesson: [
            "An emergency fund is the difference between a setback and a spiral.",
            "Target: 3 months of bare-minimum expenses (rent + food + insurance, no nights out).",
            "Keep it in a high-yield savings account (Ally / Marcus / Apple Card Savings). 4-5% currently.",
            "Don't invest it. Returns don't matter — accessibility does. Liquid = first month, growth = later money.",
            "Action: pick the bank, calculate your number, set up a $50/week auto-transfer. You'll get there."
          ]),
]

// MARK: - Squad members
struct SquadMember: Identifiable, Hashable {
    let id: String
    let name: String
    let emoji: String
    let bestReturn: Double
    let badges: [String]
}

let SQUAD: [SquadMember] = [
    SquadMember(id: "maya",  name: "Maya",  emoji: "🌸", bestReturn: 0.42, badges: ["diversifier", "streak-30"]),
    SquadMember(id: "priya", name: "Priya", emoji: "🌿", bestReturn: 0.31, badges: ["diversifier"]),
    SquadMember(id: "zoe",   name: "Zoe",   emoji: "🌊", bestReturn: 0.18, badges: []),
    SquadMember(id: "lena",  name: "Lena",  emoji: "🍯", bestReturn: -0.05, badges: ["streak-7"]),
]

// MARK: - Watchlist
struct WatchAsset: Identifiable, Hashable {
    let id: String
    let name: String
    let price: Double
    let change: Double   // pct
    let series: [Double] // sparkline
}

let WATCHLIST: [WatchAsset] = [
    WatchAsset(id: "VTI",   name: "Total US Market",
               price: 245.18, change: 0.012,
               series: [240, 241, 239, 242, 243, 245]),
    WatchAsset(id: "BND",   name: "US Bonds",
               price: 73.40, change: 0.002,
               series: [73, 73.2, 73.1, 73.3, 73.4, 73.4]),
    WatchAsset(id: "AAPL",  name: "Apple",
               price: 178.92, change: -0.008,
               series: [180, 179, 181, 180, 179, 178.92]),
    WatchAsset(id: "NVDA",  name: "NVIDIA",
               price: 412.55, change: 0.034,
               series: [395, 401, 398, 405, 410, 412.55]),
    WatchAsset(id: "TSLA",  name: "Tesla",
               price: 244.18, change: -0.021,
               series: [255, 250, 248, 246, 247, 244.18]),
    WatchAsset(id: "VXUS",  name: "International",
               price: 58.74, change: 0.006,
               series: [58, 58.2, 58.4, 58.5, 58.7, 58.74]),
]

// MARK: - Buddy accessories
struct Accessory: Identifiable, Hashable {
    let id: String
    let name: String
    let emoji: String
    let cost: Int
}

let ACCESSORIES: [Accessory] = [
    Accessory(id: "crown",    name: "Tiny crown",  emoji: "👑", cost: 100),
    Accessory(id: "scarf",    name: "Cozy scarf",  emoji: "🧣", cost: 60),
    Accessory(id: "glasses",  name: "Bookish specs", emoji: "👓", cost: 80),
    Accessory(id: "balloon",  name: "Balloon",     emoji: "🎈", cost: 40),
    Accessory(id: "rose",     name: "Single rose", emoji: "🌹", cost: 50),
    Accessory(id: "donut",    name: "Donut",       emoji: "🍩", cost: 30),
]

File: src/data/seed.ts
export interface Quest {
  id: string;
  title: string;
  pillar: 'budget' | 'invest' | 'negotiate' | 'side-hustle' | 'benefits';
  difficulty: 'starter' | 'core' | 'boss';
  reward: { coins: number; xp: number };
  blurb: string;
  emoji: string;
  duration: string;
  kind: 'lesson' | 'simulator' | 'tracker';
}

export const quests: Quest[] = [
  {
    id: 'first-paycheck',
    title: 'Decode Your First Paycheck',
    pillar: 'budget',
    difficulty: 'starter',
    reward: { coins: 80, xp: 120 },
    blurb: 'Gross vs net, FICA, federal, state — where your money actually goes before it hits your account.',
    emoji: '💸',
    duration: '5 min',
    kind: 'lesson',
  },
  {
    id: 'negotiate-internship',
    title: 'Negotiate Your Internship Offer',
    pillar: 'negotiate',
    difficulty: 'boss',
    reward: { coins: 250, xp: 400 },
    blurb: "Role-play with an AI hiring manager. Practice anchoring, counter-offers, and silence — without the stakes.",
    emoji: '👑',
    duration: '10 min',
    kind: 'simulator',
  },
  {
    id: 'unpack-benefits',
    title: 'Unpack a Benefits Package',
    pillar: 'benefits',
    difficulty: 'core',
    reward: { coins: 180, xp: 280 },
    blurb: "401(k) match, RSUs, healthcare premiums — learn what they're actually worth in dollars.",
    emoji: '🎁',
    duration: '8 min',
    kind: 'lesson',
  },
  {
    id: 'invest-first-100',
    title: 'Invest Your First $100',
    pillar: 'invest',
    difficulty: 'core',
    reward: { coins: 200, xp: 300 },
    blurb: 'ETF, index fund, individual stock — pick a path and simulate one year of compounding.',
    emoji: '📈',
    duration: '12 min',
    kind: 'lesson',
  },
  {
    id: 'side-hustle-launch',
    title: 'Launch a Side Hustle in 7 Days',
    pillar: 'side-hustle',
    difficulty: 'core',
    reward: { coins: 220, xp: 320 },
    blurb: 'From idea to first dollar — pricing, taxes, and how to know if it’s worth your time.',
    emoji: '⚡',
    duration: '7 days',
    kind: 'tracker',
  },
  {
    id: 'ask-for-raise',
    title: 'Ask for a Raise',
    pillar: 'negotiate',
    difficulty: 'boss',
    reward: { coins: 300, xp: 450 },
    blurb: 'Build the case, time it right, and run the conversation. AI plays your boss.',
    emoji: '🔥',
    duration: '15 min',
    kind: 'simulator',
  },
];

export interface SpendingCategory {
  name: string;
  spent: number;
  budget: number;
  color: string;
  emoji: string;
}

export const spending: SpendingCategory[] = [
  { name: 'Food', spent: 410, budget: 350, color: '#FF9E7D', emoji: '🍜' },
  { name: 'Transport', spent: 86, budget: 120, color: '#7BC4F0', emoji: '🚇' },
  { name: 'Fun', spent: 145, budget: 150, color: '#EA9AB2', emoji: '🎟️' },
  { name: 'Subs', spent: 38, budget: 50, color: '#A78BFA', emoji: '🎧' },
  { name: 'Savings', spent: 200, budget: 200, color: '#6FE3B3', emoji: '🪴' },
];

export const stocks = [
  { symbol: 'NVDA', name: 'NVIDIA', delta: '+1.38%', positive: true },
  { symbol: 'AAPL', name: 'Apple', delta: '+0.46%', positive: true },
  { symbol: 'TSLA', name: 'Tesla', delta: '-0.19%', positive: false },
  { symbol: 'VTI', name: 'Vanguard Total', delta: '+0.32%', positive: true },
];

export const squadMembers = [
  { name: 'Maya', emoji: '🌸', level: 6, streak: 21, you: false },
  { name: 'You', emoji: '✨', level: 4, streak: 12, you: true },
  { name: 'Priya', emoji: '🌿', level: 5, streak: 18, you: false },
  { name: 'Zoe', emoji: '🌊', level: 3, streak: 7, you: false },
  { name: 'Lena', emoji: '🍯', level: 4, streak: 9, you: false },
];

export const shopItems: { id: string; label: string; emoji: string; cost: number }[] = [
  { id: 'crown', label: 'Crown', emoji: '👑', cost: 250 },
  { id: 'glasses', label: 'Glasses', emoji: '🕶️', cost: 120 },
  { id: 'headphones', label: 'Headphones', emoji: '🎧', cost: 180 },
  { id: 'flower', label: 'Flower Crown', emoji: '🌸', cost: 200 },
  { id: 'cap', label: 'Cap', emoji: '🧢', cost: 90 },
  { id: 'sparkle', label: 'Sparkle Aura', emoji: '✨', cost: 60 },
];
